src/ripple/rpc/handlers/LedgerRequest.cpp

```diff
@@ -118,28 +119,27 @@ Json::Value doLedgerRequest (RPC::Context& context)
         ledgerHash = neededHash ? *neededHash : zero; // kludge
     }
 
-    auto ledger = ledgerMaster.getLedgerByHash (ledgerHash);
+    // Try to get the desired ledger
+    auto ledger = context.app.getInboundLedgers().acquire (
+        ledgerHash, ledgerIndex, InboundLedger::fcGENERIC);
+
+    if (context.app.config().standalone())
```

Comment:
If we unconditionally overwrite `ledger` in standalone mode, shouldn't this be:

    std::shared_ptr<Ledger const>;

    if (context.app.config().standalone())
        ledger = ledgerMaster.getLedgerByHash(ledgerHash);
    else
        ledger = context.app.getInboundLedgers().acquire (
            ledgerHash, ledgerIndex, InboundLedger::fcGENERIC);